features/filesystem/bd/FlashSimBlockDevice.cpp

```diff
@@ -0,0 +1,142 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FlashSimBlockDevice.h"
+#include "mbed_assert.h"
+#include <algorithm>
+#include <stdlib.h>
+
+static const uint8_t blank = 0xFF;
```

Comment:
Do you think we should make this configurable in the constructor? Since this variable is the point of the class.

Reply:
Good question. Would argue that this value is device specific, and that a user of either this class or a real flash block device would need to read this value via the `get_erase_value` API, hence wouldn't be interested in the predefined value.

Reply:
In that case I suggest we should add an API to block device  class to read this value from block device

Reply:
There is such an API - `get_erase_value`, and this class implements it (just returns `blank`).

Reply:
Sorry missed that.  If it is device specific and not application/user specific we can have a macro and target device based on the RAM in hardware can define this macro in targets.json `macro` section. But I am not sure if any RAM device will have erase state as 0.